src/containers/EditDonor/index.js

```diff
@@ -0,0 +1,221 @@
+import React, {Component} from "react";
+import Button from "react-bootstrap/Button";
+import FormGroup from "react-bootstrap/FormGroup";
+import Col from "react-bootstrap/Col";
+import Row from "react-bootstrap/Row";
+import Card from "react-bootstrap/Card";
+import * as actions from "./actions";
+import FormControl from "react-bootstrap/FormControl";
+import {bindActionCreators} from "redux";
+import {connect} from "react-redux";
+import Alert from "react-bootstrap/Alert";
+
+//TODO get zipcode update and institution get and update to work!  Rerender list and forms on back button
+function mapDispatchToProps(dispatch) {
+  return bindActionCreators({
+    getDonor: actions.getDonor,
+    updateDonor: actions.updateDonor,
+  }, dispatch);
+}
+
+export class EditDonor extends Component {
+  constructor(props) {
+    super(props)
+    this.myRef = React.createRef();
+    this.state = {
+      name: "",
+      phone: "",
+      email: "",
+      address: "",
+      city: "",
+      addrState: "",
+      zipCode: null,
+      institution: ""
+    }
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    this.props.getDonor(this.props.match.params.donorId).then(response => {
+      this.setState({
+        name: response.payload.donorName,
+        phone: response.payload.phone,
+        email: response.payload.email,
+        address: response.payload.address,
+        city: response.payload.city,
+        addrState: response.payload.state,
+        zipCode: response.payload.zipCode != null ? toString(response.payload.zipCode) : null,
+        institution: response.payload.institution
+      })
+    })
+  }
+
+
+
+  // validateForm() {
+  //   return this.email.length > 0 && this.password.length > 0;
+  // }
+
+
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+    const { name, email, phone, address, addrState, city, zipCode, institution } = this.state;
+    const { updateDonor } = this.props;
+    updateDonor(this.props.match.params.donorId, name, email, phone, address, addrState, city, parseInt(zipCode), institution).then(response => {
+      this.setState({ submitted: true })
+      if (response.type === 'SUCCESS') {
+        this.setState({ success: true })
+      }
+      if (response.type === 'FAILURE') {
+        this.setState({ success: false })
+      }
+      setTimeout(() => {
+        this.setState({ submitted: false });
```

Comment:
You could probably add this.props.history.push('/donors') here so that on successful creation, the page gets redirected to the donors page being consistent with the usual web app behavior. But I think you probably should also check if success=true before doing that. If the edit failed, we'd probably want the user to stay on the same page.

Reply:
will update after initial merge to master